src/pybind/mgr/smb/handler.py

```diff
@@ -836,6 +848,27 @@ def _check_share(
         raise ErrorResult(
             share, msg="path is not a valid directory in volume"
         )
+    earmark = earmark_resolver.get_earmark(
+        share.cephfs.path,
+        share.cephfs.volume,
+        share.cephfs.subvolumegroup,
+        share.cephfs.subvolume,
+    )
+    if not earmark:
+        earmark_resolver.set_earmark(
+            share.cephfs.volume,
+            share.cephfs.subvolumegroup,
+            share.cephfs.subvolume,
+            EarmarkTopScope.SMB.value,
+        )
+    else:
+        if not earmark_resolver.check_earmark(
+            earmark, EarmarkTopScope.SMB.value
+        ):
+            raise ErrorResult(
+                share,
+                msg=f"earmark is already been set by {earmark.split('.')[0]}. If you want to use this path, please clear the earmark first or set the earmark to smb",
```

Comment:
1. Long line. Please split it.
2. I would rather have the "extra help" refer to some docs or something... telling people to clear the earmark is just handing naive users a dangerous power tool without the safety instructions. For now, we can even choose to say nothing beyond 'earmark already set'. But we can discuss this more if needed.

Reply:
Yes I'm planning to add some extra help doc on how to deal with earmark & how to set/clear if already been set